build/glimmer-template-precompiler.js

```diff
@@ -0,0 +1,33 @@
+const Filter = require('broccoli-persistent-filter');
+const compiler = require('@glimmer/compiler');
+
+class GlimmerTemplatePrecompiler extends Filter {
+  constructor(inputNode, options) {
+    super(...arguments);
+    this.options = options || {};
+    this.extensions = ['hbs'];
+    this.targetExtension = 'ts';
+  }
+
+  processString(content, relativePath) {
+    let specifier = getTemplateSpecifier(this.options.rootName, relativePath);
+    return 'export default ' + compiler.precompile(content, { meta: { specifier, '<template-meta>': true } }) + ';';
+  }
+}
+
+function getTemplateSpecifier(rootName, relativePath) {
+  let path = relativePath.split('/');
+  let prefix = path.shift();
+
+  // TODO - should use module map config to be rigorous
+  if (path[path.length - 1] === 'template.hbs') {
+    path.pop();
+  }
+  if (path[0] === 'ui') {
+    path.shift();
+  }
+
+  return 'template:/' + rootName + '/' + path.join('/');
+}
+
+module.exports = GlimmerTemplatePrecompiler;
```

Comment:
BTW, we don't need `<glimmer-template>` in production builds. I spoke with @wycats and it was just a hack for making type checking of template types nominal instead of structural. It's not used at runtime.